db/db_utils.go

```diff
@@ -411,3 +412,18 @@ func AggregateWithHint(collection string, pipeline interface{}, hint interface{}
 
 	return errors.WithStack(pipe.All(out))
 }
+
+// AggregateWithMaxTime runs aggregate and specifies a max quert time which ensures the query won't go on indefinitely when the request is cancelled.
+func AggregateWithMaxTime(collection string, pipeline interface{}, out interface{}, maxTime time.Duration) error {
```

Comment:
Should we refactor the other usages of MaxTime() to use this function?

Reply:
Sounds good!